### ISSUE TITLE
nice work, did sinkhorn-attention could be applied to casual-language model?

### ISSUE THREAD

**Issue by yyht** [michaelsdr/sinkformers#1]
Hi, sinkhorn-transform is a great work, how to use sinkhorn-attention into casual language model? does it just replace bi-attention mask with casual mask for cost matrix?

attn_score = torch.matmul(q, k.transpose(-1, -2)) / np.sqrt(self.d_k)
attn_score.masked_fill_(attn_mask, -1e9)---------in general, attn_mask for bi-directional modeling attends to all tokens, if we apply to casual attention for GPT-2, we have to replace attn_mask with casual mask, when we do it, is it valid to use sinkhorn-attention?
attn_score_shape = attn_score.shape
attn_weights_soft = nn.Softmax(dim=-1)(attn_score)
# |attn_score| : (batch_size, n_heads, q_len, k_len)
attn_score = attn_score.view(-1, attn_score_shape[2], attn_score_shape[3])
sink = SinkhornDistance(1, max_iter=n_it)
